tests/unit/custom-actions-test.js

```diff
@@ -117,4 +117,66 @@ module('customAction()', function (hooks) {
     });
     assert.deepEqual(response, { adapterOptions: 'it works' });
   });
+
+  test('should handle query params', async function (assert) {
+    class UserAdapter extends RESTAdapter {
+      buildURL(modelName, id, snapshot, requestType, query) {
+        let url = super.buildURL(...arguments)
+        if (query) {
+          let params = new URLSearchParams(query)
+          url += `?${params.toString()}`;
+        }
+        return url;
+      }
+    }
+
+    this.owner.register('adapter:user', UserAdapter);
+
+    let { worker, rest, user } = await prepare(this);
+
+    worker.use(
+      rest.post('/users?order=asc', (req, res, ctx) => {
+        assert.equal(req.url.searchParams.get('order'), 'asc', 'request made to the right URL');
+        return res(ctx.json({ adapterOptions: 'it works' }));
+      })
+    );
+
+    let response = await apiAction(user, {
+      method: 'POST',
+      requestType: 'createRecord',
+      query: { order: 'asc' },
+    });
```

Comment:
```suggestion
    let response = await apiAction(user, {
      method: 'POST',
      requestType: 'createRecord',
      path: `?order=asc',
    });
```

I think we can already encode query parameters in the `path`, so I'm wondering if the additional code and complexity is really necessary.

Reply:
Oh neat, didn't know that worked.

IMHO we should allow both, since the original JSONAPI Adapter supports it. WDYT ?

Reply:
but why? It's additional complexity and code we have to send to the user and if we already have a way of doing it then IMHO we need a strong reason to add another one.

Reply:
We've got a `buildURL` that appends a query param regardless of the `query` argument. We want the query param for every call to the endpoint.

```js
  buildURL(modelName, id, snapshot, requestType, query) {
    let url = new URL(super.buildURL(modelName, id, snapshot, requestType, query));
    url.searchParams.append('key', value);
    return url.toString();
  }
```
If ember-api-actions handled query params being output from the `buildURL` it'd make adding an apiAction to the model simple.

Reply:
right, but that's a completely different feature request then... 😅

Reply:
Stick with the `path: ?order=asc` query param declaration?

Reply:
I can take a look on Monday

Reply:
Thanks.

In the meantime, I'll push an updated version of the MR without adding more arguments to the `apiAction` function API, but still handling cases where the adapter `buildURL` would return a url containing query params.

In the current version of the function, if the adapter `buildURL` from the client returns query params and a path, the given query params get smuched between baseURL and path :

```javascript
// if path === 'action?order=asc'
let baseUrl = adapter.buildURL(modelName, record.id, snapshot, requestType); 
// and if baseUrl === http://website.com?query=true
let url = path ? `${baseUrl}/${path}` : baseUrl;
// then: url === http://website.com?query=true/action?order=asc
```

Reply:
I took a close look at the `query` parameter to the `buildURL()` method, and according to the TypeScript definitions (https://github.com/emberjs/data/blob/v4.9.1/packages/adapter/addon/-private/build-url-mixin.ts#L147-L241) this parameter is only relevant for the `query` request types.

Reply:
You're right. The last version of this MR didnt use `query` anymore.

Since #151 answers our needs, this becomes redundant anyway